### DISCUSSION TITLE
Moving packages to new vender after renaming github account

### DISCUSSION THREAD

**Discussion by dbrumann** [composer/composer#9657] [Q&A / Support] [answered]
Hi,

I am about to rename a GitHub organization, let's say from `foo` to `bar`. I have a repository `foo/lib-x` that is registered on packagist under the same name. I want to mark the old package as abandoned and then use `bar/lib-x` instead.

This is the outcome I want:
 * github.com/foo will redirect to github.com/bar
 * on packagist I will have my existing releases under foo/lib-x
 * all future releases should be under bar/lib-x and should no longer affect foo/lib-x

This is how I want to approach it:
 * Rename the github-account
 * Change the package name in composer.json to the new vendor name
 * Add the package under the new vendor to packagist (with the new github repo url)
 * Abandon the old package in packagist (leave the github repo url)

Expected problems:
 * As far as I can tell github will automatically redirect to the new name and packagist will therefore still pick up new releases under the old package, is that right? Is there a way to avoid this?
 * The old releases will still be in the repository for the new vendor. Will they show up as releases there? I want to avoid this, because we might have some packages where we change class names which is a breaking change. We could use class_alias, but want to avoid this if possible. It should be a clean cut if possible.

Is there anything else that I have missed for this change?

**Comment by stof** [comment, marked as answer]
>     * The old releases will still be in the repository for the new vendor. Will they show up as releases there? I want to avoid this, because we might have some packages where we change class names which is a breaking change. We could use class_alias, but want to avoid this if possible. It should be a clean cut if possible.

I don't think there is a a way to avoid that if you simply rename the repository on github instead of creating a new one in which you don't push tags.
My suggestion for that would be to bump the major version number for packages in which you rename the classes.

**Reply by dbrumann** [reply]
Thanks, that's what I feared. We will do as you suggest regarding a major upgrade for renamed classes and maybe a minor release under the old name where we already provide the new class names via alias.

**Comment by dbrumann** [comment]
I think we found a feasible workaround for us, so that the old vendor will not get updates for new releases anymore. We will create a new repo called `lib-x-archived` in github that is a clone of the original repository in its current state (including tags) and immediately archive it. Then we will switch the repository for the old package to this archived repository.

It looks like having the old releases under the new vendor is ok for now. If we have to remove them, we would need to change the git history which is something we want to avoid to not mess with users depending on it.

**Reply by Seldaek** [reply]
Yup that's the only way to prevent new releases showing up.

**Reply by xabbuh** [reply]
Thanks for the feedback. 👍 And actually this approach works quite well and is not hard to set up.
